pod can not scheduler ,no event,kube-scheduler panic

### Area

- [X] Scheduler
- [ ] Controller
- [ ] Helm Chart
- [ ] Documents

### Other components

_No response_

### What happened?

```
- schedulerName: trimaran-scheduler
  plugins:
    score:
      disabled:
      - name: NodeResourcesBalancedAllocation
      - name: NodeResourceFit
      enabled:
       - name: LoadVariationRiskBalancing
  pluginConfig:
  - name: LoadVariationRiskBalancing
    args:
      safeVarianceMargin: 1
      safeVarianceSensitivity: 2
      metricProvider:
        type: KubernetesMetricsServer
        address: https://metrics-server.kube-system.svc.cluster.local:443
        token: eyJhbGciOiJSUzI1NiIsImtpZCI6IkQzTndMYkFBSnhLcDJLYTBER1ZwSlVxT1RBMlFYbVd2QXZLZFJzTGV6RFkifQ.eyJhdWQiOlsiaHR0cHM6Ly9rdWJlcm5ldGVzLmRlZmF1bHQuc3ZjLmNsdXN0ZXIubG9jYWwiXSwiZXhwIjoxNzM2NjYxMTc2LCJpYXQiOjE3MzY2NTc1NzYsImlzcyI6Imh0dHBzOi8va3ViZXJuZXRlcy5kZWZhdWx0LnN2Yy5jbHVzdGVyLmxvY2FsIiwianRpIjoiYTQ2ZGY0YzYtYjY0ZS00ZGU2LTllNDMtMzVhN2FiOGRkMmM3Iiwia3ViZXJuZXRlcy5pbyI6eyJuYW1lc3BhY2UiOiJrdWJlLXN5c3RlbSIsInNlcnZpY2VhY2NvdW50Ijp7Im5hbWUiOiJtZXRyaWNzLXNlcnZlciIsInVpZCI6ImU4YjQxZmY4LTZiZDAtNDdiYS1iZTc0LTkxN2RlNzdjM2RhZSJ9fSwibmJmIjoxNzM2NjU3NTc2LCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6a3ViZS1zeXN0ZW06bWV0cmljcy1zZXJ2ZXIifQ.Pm8lzu1IZHH2n4uD9NOiqG9POw1WNTcFJiGcSsydEATS7MSZDsc7UPs70RZQr3rFI2tva4u1mfy_buiOq8bpUDxx7JCgtiCUI3hTV-gybFlPACrFVNzkv8ulvXisBU24wOgHifFITeiy3KZxZZi9E3ILDMA7j6g4DdxBB3iMTk2k3QB2b9xvYCGoYS4OdMRmO8FABOcpH5vccrt_ucKGl5D9-RdIZiowerQRyYxcO0lOdPbc3mMh6j3QOzlZeg9E1q_Ez1-0UMTYR3qTcIjNaUtW8E6Ay8NG_CbYBLT6Jk4nVmSJug_XbYeDw5Dsy_0kpekHmrBkCjt0xFXaFhJR0Q
```

W0112 06:23:27.624803      12 client_config.go:664] error creating inClusterConfig, falling back to default config: open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x34ce94c]

goroutine 1 [running]:
github.com/paypal/load-watcher/pkg/watcher.(*Watcher).GetLatestWatcherMetrics(0x0, {0x3ccae5e, 0x3})
        /root/go/pkg/mod/github.com/paypal/load-watcher@v0.2.4/pkg/watcher/watcher.go:191 +0x4c
github.com/paypal/load-watcher/pkg/watcher/api.libraryClient.GetLatestWatcherMetrics({{0x0, 0x0}, 0x0})
        /root/go/pkg/mod/github.com/paypal/load-watcher@v0.2.4/pkg/watcher/api/client.go:78 +0x5b
sigs.k8s.io/scheduler-plugins/pkg/trimaran.(*Collector).updateMetrics(0xc0001368c0, {{0x3f9bbe8, 0xc000906dc0}, 0x0})
        /root/scheduler-plugins-master/pkg/trimaran/collector.go:134 +0x6e
sigs.k8s.io/scheduler-plugins/pkg/trimaran.NewCollector({{0x3f9bbe8, 0xc000906dc0}, 0x0}, 0xc0009a71a0)
        /root/scheduler-plugins-master/pkg/trimaran/collector.go:77 +0x65f
sigs.k8s.io/scheduler-plugins/pkg/trimaran/loadvariationriskbalancing.New({0x3f95e38, 0xc000919d10}, {0x3f6ffe8, 0xc0009a7180}, {0x3fbbd80, 0xc00088a488})
        /root/scheduler-plugins-master/pkg/trimaran/loadvariationriskbalancing/loadvariationriskbalancing.go:63 +0x18d
k8s.io/kubernetes/pkg/scheduler/framework/runtime.NewFramework({0x3f95e38, 0xc000919d10}, 0xc00096fc80, 0xc000c8e7c8, {0xc00060f1e0, 0xc, 0x16})
        /root/go/pkg/mod/k8s.io/kubernetes@v1.31.2/pkg/scheduler/framework/runtime/framework.go:338 +0x10c9
k8s.io/kubernetes/pkg/scheduler/profile.newProfile({0x3f95e38, 0xc000919d10}, {{0xc0001a98c0, 0x12}, 0x0, 0xc0001b1188, {0xc00040c700, 0x8, 0x8}}, 0xc00096fc80, ...)
        /root/go/pkg/mod/k8s.io/kubernetes@v1.31.2/pkg/scheduler/profile/profile.go:42 +0x174
k8s.io/kubernetes/pkg/scheduler/profile.NewMap({0x3f95e38, 0xc000919d10}, {0xc0009064c0, 0x1, 0x1}, 0xc00096fc80, 0xc00096a810, {0xc000c8ed70, 0xb, 0xb})
        /root/go/pkg/mod/k8s.io/kubernetes@v1.31.2/pkg/scheduler/profile/profile.go:55 +0x22e
k8s.io/kubernetes/pkg/scheduler.New({0x3f95e38, 0xc000919d10}, {0x3fc7348, 0xc000a00540}, {0x3fbc3b8, 0xc000136460}, {0x3f96940, 0xc0009a4d80}, 0xc00096a810, {0xc000c8f348, ...})
        /root/go/pkg/mod/k8s.io/kubernetes@v1.31.2/pkg/scheduler/scheduler.go:304 +0xd37
k8s.io/kubernetes/cmd/kube-scheduler/app.Setup({0x3f95e38, 0xc000919d10}, 0xc0005d3170, {0xc000351570, 0xe, 0xe})
        /root/go/pkg/mod/k8s.io/kubernetes@v1.31.2/cmd/kube-scheduler/app/server.go:413 +0xb49
k8s.io/kubernetes/cmd/kube-scheduler/app.runCommand(0xc0001f2308, 0xc0005d3170, {0xc000351570, 0xe, 0xe})
        /root/go/pkg/mod/k8s.io/kubernetes@v1.31.2/cmd/kube-scheduler/app/server.go:153 +0x385
k8s.io/kubernetes/cmd/kube-scheduler/app.NewSchedulerCommand.func2(0xc0001f2308, {0xc0005ff800, 0x0, 0x6})
        /root/go/pkg/mod/k8s.io/kubernetes@v1.31.2/cmd/kube-scheduler/app/server.go:103 +0x77
github.com/spf13/cobra.(*Command).execute(0xc0001f2308, {0xc000050080, 0x6, 0x6})
        /root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xfca
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001f2308)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x9d0
github.com/spf13/cobra.(*Command).Execute(0xc0001f2308)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041 +0x32
k8s.io/component-base/cli.run(0xc0001f2308)
        /root/go/pkg/mod/k8s.io/component-base@v0.31.2/cli/run.go:143 +0x36f
k8s.io/component-base/cli.Run(0xc0001f2308)
        /root/go/pkg/mod/k8s.io/component-base@v0.31.2/cli/run.go:44 +0x45
main.main()
        /root/scheduler-plugins-master/cmd/scheduler/main.go:69 +0x4c5
2025-01-12T06:24:25Z debug layer=debugger detaching
2025-01-12T06:24:25Z debug layer=debugger detaching
```

```
root@k8s-master01:/etc/kubernetes/manifests# cat kube-scheduler.yaml 
apiVersion: v1
kind: Pod
metadata:
  labels:
    component: kube-scheduler
    tier: control-plane
  name: kube-scheduler
  namespace: kube-system
spec:
  containers:
  - command:
    - dlv 
    - exec
    - /bin/scheduler
    - --log
    - --listen=0.0.0.0:40000 
    - --headless
    - --api-version=2
    #  - --continue
    - --accept-multiclient
    - --
    - --config=/etc/kubernetes/sched-cc.yaml
    - --kubeconfig=/etc/kubernetes/scheduler.conf
    - --authentication-kubeconfig=/etc/kubernetes/scheduler.conf
    - --authorization-kubeconfig=/etc/kubernetes/scheduler.conf
    - --bind-address=127.0.0.1
    - --v=2
    image: registry.cn-hangzhou.aliyuncs.com/hxpdocker/scheduler-plugins:v1.31 
    imagePullPolicy: Always
#    livenessProbe:
#      failureThreshold: 8
#      httpGet:
#        host: 127.0.0.1
#        path: /healthz
#        port: 10259
#        scheme: HTTPS
#      initialDelaySeconds: 10
#      periodSeconds: 10
#      timeoutSeconds: 15
    name: kube-scheduler
    resources:
      requests:
        cpu: 100m
#    startupProbe:
#      failureThreshold: 24
#      httpGet:
#        host: 127.0.0.1
#        path: /healthz
#        port: 10259
#        scheme: HTTPS
#      initialDelaySeconds: 10
#      periodSeconds: 10
#      timeoutSeconds: 15
    volumeMounts:
    - mountPath: /etc/kubernetes/sched-cc.yaml
      name: sched-cc
      readOnly: true
    - mountPath: /etc/kubernetes/scheduler.conf
      name: kubeconfig
      readOnly: true
  hostNetwork: true
  priority: 2000001000
  priorityClassName: system-node-critical
  securityContext:
    seccompProfile:
      type: RuntimeDefault
  volumes:
  - hostPath:
      path: /etc/kubernetes/sched-cc.yaml
      type: FileOrCreate
    name: sched-cc
  - hostPath:
      path: /etc/kubernetes/scheduler.conf
      type: FileOrCreate
    name: kubeconfig
```

```
root@k8s-master01:/etc/kubernetes# kubectl describe pod nginx-74df8bd4ff-xdzr8 
Name:             nginx-74df8bd4ff-xdzr8
Namespace:        default
Priority:         0
Service Account:  default
Node:             <none>
Labels:           app=nginx
                  pod-template-hash=74df8bd4ff
Annotations:      <none>
Status:           Pending
IP:               
IPs:              <none>
Controlled By:    ReplicaSet/nginx-74df8bd4ff
Containers:
  nginx:
    Image:      registry.cn-hangzhou.aliyuncs.com/hxpdocker/nginx:latest
    Port:       <none>
    Host Port:  <none>
    Limits:
      cpu:  300m
    Requests:
      cpu:        300m
    Environment:  <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-fhmcf (ro)
Volumes:
  kube-api-access-fhmcf:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:                      <none>
root@k8s-master01:/etc/kubernetes# kubectl get pod -owide
NAME                     READY   STATUS    RESTARTS   AGE     IP       NODE     NOMINATED NODE   READINESS GATES
nginx-74df8bd4ff-xdzr8   0/1     Pending   0          5m52s   <none>   <none>   <none>           <none>
```

```
cat << EOF | kubectl apply -f -
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      name: nginx
      labels:
        app: nginx
    spec:
      schedulerName: trimaran-scheduler
      containers:
        - name: nginx
          image: registry.cn-hangzhou.aliyuncs.com/hxpdocker/nginx:latest
          resources:
            limits:
              cpu: 0.3
            requests:
              cpu: 0.3
EOF
```



### What did you expect to happen?

pod can scheduler

### How can we reproduce it (as minimally and precisely as possible)?

_No response_

### Anything else we need to know?

_No response_

### Kubernetes version

<details>
1.31

</details>


### Scheduler Plugins version

<details>
master
</details>

--- a/pkg/trimaran/collector.go
+++ b/pkg/trimaran/collector.go
@@ -56,9 +56,12 @@ func NewCollector(logger klog.Logger, trimaranSpec *pluginConfig.TrimaranSpec) (
 	logger.V(4).Info("Using TrimaranSpec", "type", trimaranSpec.MetricProvider.Type,
 		"address", trimaranSpec.MetricProvider.Address, "watcher", trimaranSpec.WatcherAddress)
 
-	var client loadwatcherapi.Client
+	var (
+		client loadwatcherapi.Client
+		err    error
+	)
 	if trimaranSpec.WatcherAddress != "" {
-		client, _ = loadwatcherapi.NewServiceClient(trimaranSpec.WatcherAddress)
+		client, err = loadwatcherapi.NewServiceClient(trimaranSpec.WatcherAddress)
 	} else {
 		opts := watcher.MetricsProviderOpts{
 			Name:               string(trimaranSpec.MetricProvider.Type),
@@ -66,7 +69,11 @@ func NewCollector(logger klog.Logger, trimaranSpec *pluginConfig.TrimaranSpec) (
 			AuthToken:          trimaranSpec.MetricProvider.Token,
 			InsecureSkipVerify: trimaranSpec.MetricProvider.InsecureSkipVerify,
 		}
-		client, _ = loadwatcherapi.NewLibraryClient(opts)
+		client, err = loadwatcherapi.NewLibraryClient(opts)
+	}
+
+	if err != nil {
+		return nil, err
 	}
 
 	collector := &Collector{
@@ -74,7 +81,7 @@ func NewCollector(logger klog.Logger, trimaranSpec *pluginConfig.TrimaranSpec) (
 	}
 
 	// populate metrics before returning
-	err := collector.updateMetrics(logger)
+	err = collector.updateMetrics(logger)
 	if err != nil {
 		logger.Error(err, "Unable to populate metrics initially")
 	}

